### PR TITLE
Re design

### PR DESCRIPTION
**Pull request by hilashay**

### CODE REVIEW

File: src/Utils/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const formatKeyAndValue = (key: any) => {
```
Comment (idanhaviv) [inline src/Utils/utils.tsx]: rename function to a descriptive name - that explain what it does

Comment (idanhaviv) [reply src/Utils/utils.tsx]: and replace any with a type

File: src/Utils/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  console.log("key.length", key);
+  if (typeof key === "number") {
+    key.toString();
```
Comment (idanhaviv) [inline src/Utils/utils.tsx:3]: why is this needed?

File: src/Utils/utils.tsx (hunk 1 above)
Comment (idanhaviv) [reply src/Utils/utils.tsx]: in terms of safety of the code - currently you accept 'any' as a type, so this function can be called with any type, and if you reach the end of the function and the key type won't have a charAt method (like you previously had when you called it with a number) it would throw an error.

so, you can protect yourself in 2 ways
1. Typescript - limit the key type to what you expect
2. make sure you cover all the possible scenarios in code. you can do it by checking the type is a string before you return the last statement; or, use a switch statement of the `typeof key` with a default behavior 

lets review both implementations (using a switch and not using a switch)

File: src/Utils/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    return key;
+  }
+  return key?.charAt(0).toUpperCase() + key?.slice(1).replace(/[A-Z]/g, " $&").trim();
+};
+
+export const FontType = () => {
+  return (
+    <style> @import url('https://fonts.googleapis.com/css2?family=Dosis&display=swap'); </style>
+  );
+};
```
Comment (idanhaviv) [inline src/Utils/utils.tsx]: This way is not standard for importing fonts
checkout https://blog.greenroots.info/3-quick-ways-to-add-fonts-to-your-react-app and use your preferred way

File: src/components/General.styled.ts [hunk 4]
```diff
@@ -6,18 +6,20 @@ export const SectionContainer = styled.div`
 `;
 
 export const Button = styled.button`
```
Comment (idanhaviv) [inline src/components/General.styled.ts:4]: each file should export one component, and the file name should be the same as the component name followed by `.tsx`
separate to 3 files

File: src/modules/About/About.tsx [hunk 5]
```diff
@@ -23,7 +23,7 @@ const About = () => {
 export default About;
 
 const Container = styled(SectionContainer)`
-  background-color: #fff7f4;
+  background: #eae0de;
```
Comment (idanhaviv) [inline src/modules/About/About.tsx]: I searched your code for `#eae0de` and i see you're using it in 3 places. DRY (https://www.digitalocean.com/community/tutorials/what-is-dry-development)

create a `Colors.ts` file that contains all the colors you use in the app in use that everywhere instead of static color values. 
give the colors a meaningful name and don't forget constant letter casing (e.g. BACKGROUND, PRIMARY, SECONDARY)

Comment (idanhaviv) [reply src/modules/About/About.tsx]: note - you should apply this change to all the colors in the project not just this file

File: src/modules/About/Cards/Card.tsx [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import styled from "styled-components";
 import { CardProps } from "./Cards";
 
-const Card = ({ imageUrl, text }: CardProps) => (
+const Card = ({ image, title, text }: CardProps) => (
```
Comment (idanhaviv) [inline src/modules/About/Cards/Card.tsx]: 1. i think imageUrl is a better name, since it's more descriptive
2. keep a component's props interface (CardProps) near the component itself, so it should be declared on this file

File: src/modules/About/Cards/Cards.tsx [hunk 7]
```diff
@@ -1,33 +1,38 @@
...
-    text: "Lorem ipsum dolor sit amet consectetur adipisicing elit. enim.",
+    image: whoIsItFor,
+    title: "Who is it for",
+    text: "the service is suitable for people that don't like to do shopping, don't have time to do shopping, don't understand how this 'Shopping Online' thing work and who want to feel and look great!",
   },
 ];
 
 const Cards = () => {
   const renderCards = () =>
-    cards.map(({ imageUrl, text }, index) => <Card imageUrl={imageUrl} text={text} key={index} />);
```
Comment (idanhaviv) [inline src/modules/About/Cards/Cards.tsx:30]: don't use separate render functions (`renderCards`) just put it inline in the return statement (`<Container>....</...>`)

File: src/modules/About/Description.tsx [hunk 8]
```diff
@@ -1,24 +1,37 @@
...
+          me.
+        </Text>
+        <Text>
+          So I thought, why not be that person for other people who require this type of service?
+          Making people look and feel good about themselves make me feel satisfied, and I believe it
+          is a need that is still underserved on the needs list.
+        </Text>
       </Text>
-      <Button>Dress me!</Button>
+      <AboutButton onClick={() => navigate("/form-section")}>Dress me</AboutButton>
```
Comment (idanhaviv) [inline src/modules/About/Description.tsx]: `AboutButton` is a confusing name - since what it does is navigate to the Dress me form. why do you call it like that?

File: src/modules/About/Description.tsx [hunk 9]
```diff
@@ -1,24 +1,37 @@
+import { useNavigate } from "react-router-dom";
 import styled from "styled-components";
 import { Button, Span } from "../../components/General.styled";
 
 const Description = () => {
+  const navigate = useNavigate();
```
Comment (idanhaviv) [inline src/modules/About/Description.tsx]: it's important to separate responsibilities and concerns (אחריות) in software development. so that each module/function/component will have a single concern (as much as possible) and definitely try not to mix with other concerns. https://en.wikipedia.org/wiki/Single-responsibility_principle
so when it comes to React Components, we try to separate 'smart' components that deal with behavior and logic, and 'dumb' components that are in charge of UI.
in this case, when i'm thinking about the `Description` component, i think it makes sense that it will be a 'dumb' component in charge of rendering some description about you (maybe not a description? maybe introduction? think about it)
so when we're adding the navigate to it we make it a bit 'smarter' and when it uses it to navigate to the dressme page when the button is clicked, it has knowledge about a behavior that's related to the app as a whole. making it super smart.

how can this be solved?
separating the concerns. 
instead of Description deciding what happens when the button is clicked, it will get that decision as a prop, and we'll leave that decision to a component higher up the tree, maybe the HomePage itself.
so i would pass this function `() => navigate("/form-section")}` as a prop all the way from HomePage.
how would you name that prop? something that's related to what should the Description and About components know about - maybe `onClickDressMeButton`

File: src/modules/Brands/Brands.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    <Container id="brands-section">
+      <Title>Let me find your faves from 1000+ brands & styles</Title>
+      <Logos />
+    </Container>
+  );
+};
+
+export default Brands;
+
+const Container = styled(SectionContainer)`
```
Comment (idanhaviv) [inline src/modules/Brands/Brands.tsx:17]: I like SectionContainer! nice reusability!!

File: src/modules/Brands/Logos.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import styled from "styled-components";
+
+const importAll = (r: any) => {
+  let images: any = {};
+  r.keys().map((item: any) => {
+    images[item.replace("./", "")] = r(item);
+  });
+  return images;
+};
+const images = importAll(require.context("../../images/brands", false, /\.(png|jpe?g|svg)$/));
```
Comment (idanhaviv) [inline src/modules/Brands/Logos.tsx]: make sure you know how to explain this code

File: src/modules/DressMe/DressMe.tsx [hunk 12]
```diff
@@ -79,4 +97,5 @@ const FormButton = styled(Button)`
   width: 150px;
   font-size: 15px;
   align-self: end;
+  bottom: -80px;
```
Comment (idanhaviv) [inline src/modules/DressMe/DressMe.tsx:105]: what does this do?

File: src/modules/DressMe/DressMe.tsx [hunk 13]
```diff
@@ -1,29 +1,39 @@
-import { Form, Formik } from "formik";
-import { useEffect } from "react";
+import { Form, Formik, useFormik } from "formik";
+import { useContext, useEffect } from "react";
```
Comment (idanhaviv) [inline src/modules/DressMe/DressMe.tsx]: remove unused imports (option+shift+O)

File: src/modules/DressMe/PersonalDeatails/PersonalDeatails.tsx [hunk 14]
```diff
@@ -25,6 +25,7 @@ export default PersonalDetails;
 const Container = styled.div`
   display: flex;
   flex-direction: column;
+  // background: red;
```
Comment (idanhaviv) [inline src/modules/DressMe/PersonalDeatails/PersonalDeatails.tsx]: remove comments

File: src/modules/DressMe/TextArea.tsx [hunk 15]
```diff
@@ -1,9 +1,17 @@
+import { useField } from "formik";
 import styled from "styled-components";
 
 const TextArea = () => {
+  const [value, meta, teztAreaHelpers] = useField("other");
```
Comment (idanhaviv) [inline src/modules/DressMe/TextArea.tsx]: 1. remove unused variables - value and meta
2. typo in `teztAreaHelpers`
3. i would rename `teztAreaHelpers` to something more meaningful. it's not the helpers of the Text Area, it's the helpers of the field with the `other` name. maybe `otherFieldHelpers` will make more sense. 
also, since it's the only field you have in this scope (the TextArea component scope) maybe i would just name it `helpers`
4. another improvement - you're only using `setValue` method from the helpers, can you extract it somehow without declaring the teztAreaHelpers variable?

File: src/modules/DressMe/UserSizeSelection.tsx [hunk 16]
```diff
@@ -39,3 +45,7 @@ const Select = styled.select`
   border-color: white;
   box-shadow: 0 4px 8px 0 rgb(0 0 0 / 10%), 0 6px 20px 0 rgb(0 0 0 / 1%);
 `;
+
+function setFieldValue(arg0: string, value: any): void {
```
Comment (idanhaviv) [inline src/modules/DressMe/UserSizeSelection.tsx]: what's this function?

you probably used Intellisense (what's helping you with importing stuff that aren't imported yet) and you accidentally chose 'create a function with this name' or something like that

Comment (idanhaviv) [reply src/modules/DressMe/UserSizeSelection.tsx]: but! my review is that you should review your PR before submitting it, so you should note this function shouldn't be here 😊

File: src/modules/Footer/Social/Icons.tsx [hunk 17]
```diff
@@ -1,27 +1,36 @@
...
   {
     ImageUrl: "https://i.pinimg.com/564x/29/f7/5c/29f75cdb0cc4539f957c2b760a8e4906.jpg",
     alt: "whatsapp",
+    urlAdress: "https://wa.me/+972526506130",
   },
 ];
 
 const Icons = () => {
   const renderIcons = () =>
-    icons.map(({ ImageUrl, alt }, index) => <Image key={index} src={ImageUrl} alt={alt} />);
```
Comment (idanhaviv) [inline src/modules/Footer/Social/Icons.tsx:24]: remove renderIcons function

File: src/modules/Footer/Social/Social.tsx [hunk 18]
```diff
@@ -9,10 +9,8 @@ bibendum.`;
 const Social = () => {
   return (
     <Container>
```
Comment (idanhaviv) [inline src/modules/Footer/Social/Social.tsx:6]: חיים שלי את!
ראיתי שכתבת לעצמך הערה עם trade off למטה ❤️
איך היא אוהבת טריידאופים...

please remove all the unused stuff from this file (and others)

File: src/modules/Header/Header.tsx [hunk 19]
```diff
@@ -1,11 +1,14 @@
...
 import styled from "styled-components";
 import logo from "../../../src/images/logo.png";
 import { Nav } from "./Nav";
 
 const Header = () => {
+  const navigate = useNavigate();
   return (
     <Container>
-      <Image src={logo} alt="error" />
+      <Image src={logo} alt="error" onClick={() => navigate("/")} />
```
Comment (idanhaviv) [inline src/modules/Header/Header.tsx:11]: same comment as in Description - pass that function as a prop named `onClickLogo` from outside. the Header shouldn't be deciding the behavior when the logo is clicked (separation of concerns)

File: src/modules/Success/SuccessPage.tsx [hunk 20]
```diff
@@ -0,0 +1,94 @@
+import styled from "styled-components";
+import { SectionContainer } from "../../components/General.styled";
+import { useContext, useEffect, useState } from "react";
+import { MyContext } from "../../providers/FormValuesContext";
+import { formatKeyAndValue } from "../../Utils/utils";
+import { Span } from "../../components/General.styled";
+
+const SuccessPage = () => {
+  const values = useContext<any>(MyContext);
```
Comment (idanhaviv) [inline src/modules/Success/SuccessPage.tsx:9]: great work with the context!
now improve that - export a custom hook on `FormValuesContext.tsx` the return this, instead of using `useContext` directly here.
this also relates to separation of concerns - the SuccessPage should be aware of the framework that provides it with the data (React Context) it should only be aware with `what` it needs. so i would name the hook `useFormDetails` so that its consumer (the SuccessPage) is only familiar with `what` the data it's consuming and not `how` it's implemented

File: src/Utils/utils.tsx (hunk 2 above)
Comment (hilashay) [reply src/Utils/utils.tsx:3]: exactly because the input could be also "number" type and then it doesn't have the function "charAt". Am I wrong?

File: src/modules/About/Description.tsx (hunk 9 above)
Comment (hilashay) [reply src/modules/About/Description.tsx]: Screen Shot 2022-12-19 at 14 52 15

not sure you'll see the screenshot but I couldn't don't that by your interactions.
Maybe we should try together

File: src/modules/Brands/Logos.tsx (hunk 11 above)
Comment (hilashay) [reply src/modules/Brands/Logos.tsx]: start explaining in google docs file but i'm stuck.

File: src/modules/DressMe/DressMe.tsx (hunk 12 above)
Comment (hilashay) [reply src/modules/DressMe/DressMe.tsx:105]: that's a solution i found on internet to resolve the space between main to footer.

File: src/modules/DressMe/DressMe.tsx (hunk 13 above)
Comment (hilashay) [reply src/modules/DressMe/DressMe.tsx]: just remove useContext, validate is necessary just in comment for not validate any time I filled the form

File: src/modules/DressMe/TextArea.tsx (hunk 15 above)
Comment (hilashay) [reply src/modules/DressMe/TextArea.tsx]: 1. i thought its important because helpers is always at third place. no?

Comment (hilashay) [reply src/modules/DressMe/TextArea.tsx]: I also try now and I couldn't just use setValue with destructuring   like this:
[ { setValue }] = useField("other");
It works just like this: 
  const [value, meta, { setValue }] = useField("other");

like I meant in one section.

File: src/modules/Footer/Social/Social.tsx (hunk 18 above)
Comment (hilashay) [reply src/modules/Footer/Social/Social.tsx:6]: I like the most this conversation :) hearts and emojis

File: src/modules/Success/SuccessPage.tsx (hunk 20 above)
Comment (hilashay) [reply src/modules/Success/SuccessPage.tsx:9]: Also don't understand. I'll be happy we'll do this together :)

File: src/Utils/utils.tsx (hunk 2 above)
Comment (idanhaviv) [reply src/Utils/utils.tsx:3]: check this (run code, console log)

File: src/Utils/utils.tsx (hunk 1 above)
Comment (idanhaviv) [reply src/Utils/utils.tsx]: check the function with the switch statement with types other than string or number (run it, log, check)
